sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -95,6 +95,12 @@ abstract class LogicalPlan extends QueryPlan[LogicalPlan] with Logging {
   }
 
   /**
+   * Estimate complete [[Statistics]] for this plan. Each plan need to override this to get
+   * complete statistics.
+   */
+  def completeStats: Statistics = statistics
```

Comment:
In this way, we can separate cbo path from the original path when we add a conf for cbo in following tasks. If cbo is open, we use completeStats, if it is off, we still use the original statistics.
When cbo is mature in the future, we can remove the original statistics and use the new completeStats instead.